omero/developers/Model/KeyValuePairs.txt

```diff
@@ -0,0 +1,90 @@
+Key-value pairs
+===============
+
+Ongoing advancements in microscopic techniques, analysis
+systems, and other areas for which OME attempts to provide
+metadata exchange require a certain flexibility in the OME
+model. There is a need to be able to store instrument
+configuration, script parameters, and similar for later use.
+
+Basic text representations are too difficult to parse to be
+of significant use. An XML format can be more easily parsed,
+but a single format would have to be agreed upon by all
+users. It seemed most useful to add particular extension
+points to the model for which no consensus on a data format
+has been reached, but where more structure than just text is
+needed. A `hash map` was the most likely candidate for this
+data.
+
+However, rather than limit this to the programming concept
+of a `hash map` or `associative array`, OME maps are defined
+as a **"non-unique ordered list of key-value pairs"**.
```

Comment:
I'm trying to think how to phrase this so it's clear that it isn't the list that's importantly non-unique, but the keys or pairs.